From f96ecec5cf2f0a0cdf526fafc9cf47d604b1d940 Mon Sep 17 00:00:00 2001
From: Jason Perlow <jperlow@gmail.com>
Date: Thu, 9 Jul 2026 20:05:41 -0400
Subject: [PATCH 1/6] misc: armchina-npu: fix pm_runtime_put void return for
 7.2

pm_runtime_put() changed from int to void in kernel 7.0.x.
Remove ret= assignments from sky1_npu_pm_runtime_put() and
sky1_npu_runtime_suspend() to fix compile error on 7.2.

Source: linux-cix-sky1-next-7.1/patches-7.1/2005-armchina-npu-fix-pm-runtime-put-void-7.0.9.patch
Applies unchanged.
---
 drivers/misc/armchina-npu/sky1/sky1.c | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/drivers/misc/armchina-npu/sky1/sky1.c b/drivers/misc/armchina-npu/sky1/sky1.c
index 80daab087e99..cce42604e1ac 100644
--- a/drivers/misc/armchina-npu/sky1/sky1.c
+++ b/drivers/misc/armchina-npu/sky1/sky1.c
@@ -280,13 +280,8 @@ int sky1_npu_pm_runtime_get_sync(struct device *dev)
 int sky1_npu_pm_runtime_put(struct device *dev)
 {
 #ifdef CONFIG_PM
-	int ret = 0;
-
-	ret = pm_runtime_put(dev);
-	if (ret < 0)
-		dev_err(dev, "PM runtime put failed! ret=%d", ret);
-
-	return ret;
+	pm_runtime_put(dev);
+	return 0;
 #else /* !CONFIG_PM  */
 	return 0;
 #endif /* CONFIG_PM */
@@ -487,11 +482,7 @@ static int sky1_npu_runtime_suspend(struct device *dev)
 
 	if (has_acpi_companion(dev)) {
 		for (int i = 0; i < CIX_NPU_PD_NUM; i++) {
-			ret = pm_runtime_put(cix_aipu_priv->pd_core[i]);
-			if (ret < 0) {
-				dev_err(cix_aipu_priv->pd_core[i], "NPU core PM runtime put failed! ret=%d", ret);
-				return ret;
-			}
+			pm_runtime_put(cix_aipu_priv->pd_core[i]);
 		}
 	}
 
-- 
2.51.0

